src/transformers/generation/beam_search.py

```diff
@@ -279,15 +279,14 @@ def process(
                     else:
                         beam_index = None
 
-                    # skip the corner case where the very first generated token is eos_token
-                    if decoder_prompt_len == input_ids.shape[-1]:
-                        continue
+                    # add up to the length which the next_scores is calculated on
+                    generated_len = input_ids[batch_beam_idx].shape[-1] + 1 - decoder_prompt_len
```

Comment:
If I'm not mistaken, this is the same as `cur_len` (L228). I'd suggest renaming `cur_len` into `generated_len`, which is more representative of the variable contents!

Reply:
Yes, the `input_ids[batch_beam_idx].shape[-1]` should be same for each `batch_beam_idx`, so we can simply use `cur_len` here.